anagram-test.cpp

```diff
@@ -22,3 +22,9 @@ TEST_CASE("selects anagrams of a word") {
         {"stream", "something", "maters"});
     REQUIRE(selection == std::vector<std::string>{"stream", "maters"});
 }
+
```

Comment:
Nice test cases added 👍 
Any reason for not testing the second example in Wikipedia (rich-chosen...)?

Reply:
Yes, sorry I missed out the example. I have added a test case for this kind of example as well.